### PR TITLE
Add tests for DisplayUrlApi

### PR DESCRIPTION
**Pull request by ampuscas**
related to #1053

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/plugins/DisplayUrlApiTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.PluginManager;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.jenkinsci.test.acceptance.update_center.PluginSpec;
+import org.jenkinsci.test.acceptance.utils.mail.MailhogProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.internal.AssumptionViolatedException;
```
Comment (NotMyFault) [inline src/test/java/plugins/DisplayUrlApiTest.java:22]:
```suggestion
import org.junit.AssumptionViolatedException;
```

File: src/test/java/plugins/DisplayUrlApiTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                "print \";\" + DisplayURLProvider.get().getChangesURL(build)", jobName).split(";");
+    }
+
+    private void assertRedirect(String s) throws IOException {
+        CloseableHttpResponse response = null;
+        HttpGet get = new HttpGet(s.trim());
+        try {
+            response = (CloseableHttpResponse) client.execute(get);
+            int responseCode = response.getStatusLine().getStatusCode();
+            assertTrue(String.format("URL %s is not reachable", s.trim()), responseCode == 200);
```
Comment (NotMyFault) [inline src/test/java/plugins/DisplayUrlApiTest.java:133]:
```suggestion
            assertEquals(String.format("URL %s is not reachable", s.trim()), 200, responseCode);
```
Thoughts?

**Comment by rsandell** [comment]
It looks like you merged AbstractPipelineTest instead of using the upstream version.
There are added lines in the class that shouldn't be there.

**Comment by ampuscas** [comment]
> It looks like you merged AbstractPipelineTest instead of using the upstream version. There are added lines in the class that shouldn't be there.

Yes, I have modified it :)

**Review by NotMyFault — APPROVED** [review]
Thanks!

File: src/test/java/plugins/DisplayUrlApiTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
+package plugins;
```
Comment (rsandell) [inline src/test/java/plugins/DisplayUrlApiTest.java:1]: Missing license and copyright header.

File: src/test/java/plugins/DisplayUrlApiTest.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            throw new AssumptionViolatedException("Test redirect with Blue Ocean has already be tested by other test on this suite");
+        }
+    }
+
+
+    private String scriptForMailerPipeline(boolean makeFail) {
+        String status = makeFail ? "FAILURE" : "SUCCESS";
+        return String.format("node {\n" +
+                "    try {\n" +
+                "        sh 'exit 1'\n" +
```
Comment (jtnord) [inline src/test/java/plugins/DisplayUrlApiTest.java:107]: this assumes you will always be on a `linux node` which is an incorrect assumption to make.

`sh' exit 1'`  is just to fail the build - but I think you should be able to do the exact same withouth any `node` by just calling [`error`](https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#error-error-signal)

**Pull request merged**